test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetIVsApiTests/NuGetPackageSigningTestCase.cs

```diff
@@ -0,0 +1,412 @@
+                    }
+
+                    // Wait for cert to expire
+                    Thread.Sleep(15000);
+                }).GetAwaiter().GetResult();
+
+                var nugetTestService = GetNuGetTestService();
+                Assert.True(nugetTestService.EnsurePackageManagerConsoleIsOpen());
+
+                var nugetConsole = nugetTestService.GetPackageManagerConsole(project.Name);
+
+                Assert.True(nugetConsole.InstallPackageFromPMC(packageName, packageVersion));
+                Assert.True(Utils.IsPackageInstalled(nugetConsole, project.FullPath, packageName, packageVersion));
+                project.Build();
+                Assert.True(VisualStudio.HasNoErrorsInErrorList());
+                Assert.True(VisualStudio.HasNoErrorsInOutputWindows());
+
+                nugetConsole.Clear();
+                solutionService.Save();
+            }
+        }
+
+        [CIOnlyNuGetWpfTheory]
+        [InlineData(ProjectTemplate.ClassLibrary)]
+        [InlineData(ProjectTemplate.NetCoreConsoleApp)]
+        [InlineData(ProjectTemplate.NetStandardClassLib)]
+        public void InstallSignedTamperedPackageFromPMCAndFail(ProjectTemplate projectTemplate)
+        {
+            var packageName = "TestPackage";
+            var packageVersion = "1.0.0";
+
+            using (var pathContext = new SimpleTestPathContext())
+            using (var testCertificate = new X509Certificate2(_trustedTestCert.Source.Cert))
+            {
+                // Arrange
+                EnsureVisualStudioHost();
+                var solutionService = VisualStudio.Get<SolutionService>();
+
+                solutionService.CreateEmptySolution("TestSolution", pathContext.SolutionRoot);
+                var project = solutionService.AddProject(ProjectLanguage.CSharp, projectTemplate, ProjectTargetFramework.V46, "TestProject");
+                project.Build();
+
+                Task.Run(async () =>
+                {
+                    var packageToSign = Utils.CreatePackage(packageName, packageVersion);
+                    var signedPackagePath = await SignedArchiveTestUtility.CreateSignedPackageAsync(testCertificate, packageToSign, pathContext.PackageSource, packageToSign.PackageName);
+                    SignedArchiveTestUtility.TamperWithPackage(signedPackagePath);
+                }).GetAwaiter().GetResult();
+
+                var nugetTestService = GetNuGetTestService();
+                Assert.True(nugetTestService.EnsurePackageManagerConsoleIsOpen());
+
+                var nugetConsole = nugetTestService.GetPackageManagerConsole(project.UniqueName);
+
+                Assert.True(nugetConsole.InstallPackageFromPMC(packageName, packageVersion));
+                project.Build();
+
+                Assert.False(Utils.IsPackageInstalled(nugetConsole, project.FullPath, packageName, packageVersion));
```

Comment:
we should also confirm that installation failed because signed package was tampered.